app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java

```diff
@@ -100,6 +99,28 @@ public void updateDate(int id, String date) {
         });
     }
 
+    public List<String> getConnection(int id) {
```

Comment:
Should this be returning a `Connection` object instead of a `List` of its fields?

Reply:
This function should probably be async, so you can wrap it in a `Single`. So the method would look like:
```java
public Single<Connection> getConnection(int id) { 
    return Single.fromCallable(() -> {
        ...
    })
}
```

Reply:
I had initially tried returning it as a connection but the insertConnection method uses a bunch of strings instead.

Reply:
We should change `insertConnection` to take in a `Connection` then.